autotest/utilities/test_gdalalg_raster_edit.py

```diff
@@ -134,6 +134,57 @@ def test_gdalalg_raster_edit_metadata(tmp_vsimem):
         assert ds.GetMetadata() == {"AREA_OR_POINT": "Area", "bar": "baz"}
 
 
+def test_gdalalg_raster_edit_stats():
+
+    src_ds = gdal.GetDriverByName("MEM").Create("", 2, 1)
+    src_ds.GetRasterBand(1).WriteRaster(0, 0, 2, 1, b"\x01\x02")
+
+    alg = get_edit_alg()
+    alg["dataset"] = src_ds
+    alg["stats"] = True
+    assert alg.Run()
+
+    assert src_ds.GetRasterBand(1).GetMetadataItem("STATISTICS_MINIMUM") == "1"
+    assert src_ds.GetRasterBand(1).GetMetadataItem("STATISTICS_MAXIMUM") == "2"
+    assert src_ds.GetRasterBand(1).GetMetadataItem("STATISTICS_MEAN") == "1.5"
+    assert src_ds.GetRasterBand(1).GetMetadataItem("STATISTICS_STDDEV") == "0.5"
+
+
+def test_gdalalg_raster_edit_approx_stats():
+
+    src_ds = gdal.GetDriverByName("MEM").Create("", 2, 1)
+    src_ds.GetRasterBand(1).WriteRaster(0, 0, 2, 1, b"\x01\x02")
+
+    alg = get_edit_alg()
+    alg["dataset"] = src_ds
+    alg["approx-stats"] = True
+    assert alg.Run()
+
+    assert src_ds.GetRasterBand(1).GetMetadataItem("STATISTICS_MINIMUM") == "1"
+    assert src_ds.GetRasterBand(1).GetMetadataItem("STATISTICS_MAXIMUM") == "2"
+    assert src_ds.GetRasterBand(1).GetMetadataItem("STATISTICS_MEAN") == "1.5"
+    assert src_ds.GetRasterBand(1).GetMetadataItem("STATISTICS_STDDEV") == "0.5"
+
+
+def test_gdalalg_raster_edit_hist(tmp_vsimem):
+
+    tmp_filename = tmp_vsimem / "out.tif"
+    with gdal.GetDriverByName("GTiff").Create(tmp_filename, 2, 1) as ds:
+        ds.GetRasterBand(1).WriteRaster(0, 0, 2, 1, b"\x01\x02")
+
+    with gdal.Open(tmp_filename) as ds:
+        alg = get_edit_alg()
+        alg["dataset"] = ds
+        alg["hist"] = True
+        alg["auxiliary"] = True
+        assert alg.Run()
+        assert alg.Finalize()
+
+    assert gdal.VSIStatL(str(tmp_filename) + ".aux.xml") is not None
+    with gdal.VSIFile(str(tmp_filename) + ".aux.xml", "rb") as f:
+        assert b"<Histograms>" in f.read()
+
```

Comment:
Wondering if there happens to be conflicting internal metada and PAM metadata, and gdal raster exit is used to edit either the internal one or the PAM one. So is is possible to minimize the confusion? Maybe tell user somehow that there is a conflict.

Reply:
This is a larger topic for other items (nodata, CRS, metadata, etc..). And such warnings about conflicts should be emitted at the driver level. I won't address that